pkg/controller/configurationpolicy/configurationpolicy_controller.go

```diff
@@ -1230,34 +1230,16 @@ func mergeSpecsHelper(templateVal, existingVal interface{}, ctype string) interf
 			//if one field is a list and the other isn't, don't bother merging
 			return templateVal
 		}
-		if len(existingVal) > len(templateVal) {
-			//if there are more values in the existing object than the template and our complianceType is musthave,
-			//we need to merge in the extra data in the existing object to do a proper compare
-			if ctype != "mustonlyhave" {
-				return mergeArrays(templateVal, existingVal, ctype)
-			}
-			return templateVal
-		}
 		//if existing value is shorter than the template value, no merge needed since it will always be a mismatch
 		if len(existingVal) < len(templateVal) {
 			return templateVal
 		}
 		if len(existingVal) > 0 {
-			//if there are an equal amount of maps in the template and existing list, recurse on each one to merge the
-			//extra data from the existing object in
-			_, ok := existingVal[0].(map[string]interface{})
-			if ok {
-				for idx, v2 := range existingVal {
-					v1 := templateVal[idx]
-					templateVal[idx] = mergeSpecsHelper(v1, v2, ctype)
-				}
-			} else {
-				if ctype != "mustonlyhave" {
-					return mergeArrays(templateVal, existingVal, ctype)
-				}
-				return templateVal
+			//if there are more values in the existing object than the template and our complianceType is musthave,
+			//we need to merge in the extra data in the existing object to do a proper compare
+			if ctype != "mustonlyhave" {
+				return mergeArrays(templateVal, existingVal, ctype)
```

Comment:
The first thing `mergeArrays` does is check if `ctype == "mustonlyhave"`. It's not an error to have the check here, but it would simplify the flow to always return `mergeArrays`, and rely on the fact that it's handled there.